tracer/src/instruction/format/format_virtual_right_shift_i.rs

```diff
@@ -13,19 +13,25 @@ pub struct FormatVirtualRightShiftI {
     pub imm: u64,
 }
 
-#[derive(Default, Debug, Copy, Clone, Serialize, Deserialize, PartialEq)]
+#[derive(Debug, Copy, Clone, Serialize, Deserialize, PartialEq)]
 pub struct RegisterStateFormatVirtualI {
     pub rd: (u64, u64), // (old_value, new_value)
     pub rs1: u64,
 }
 
+impl Default for RegisterStateFormatVirtualI {
+    fn default() -> Self {
+        Self { rd: (0, 0), rs1: 1 }
+    }
+}
+
 impl InstructionRegisterState for RegisterStateFormatVirtualI {
     #[cfg(any(feature = "test-utils", test))]
-    fn random(rng: &mut rand::rngs::StdRng) -> Self {
+    fn random(rng: &mut rand::rngs::StdRng, operands: &NormalizedOperands) -> Self {
         use rand::RngCore;
         Self {
             rd: (rng.next_u64(), rng.next_u64()),
-            rs1: rng.next_u64(),
+            rs1: if operands.rs1 == 0 { 0 } else { rng.next_u64() },
```

Comment:
shouldn't we have a case where `if operands.rs1 == operands.rd`, then the rs1 value equals the old rd value?
this applies to some of the other formats as well